### PR TITLE
MLE-27788 add tests for dynamic host api changes

### PR DESCRIPTION
**Pull request by vitalykorolev**
### Description
This PR is based on https://github.com/marklogic/marklogic-docker/pull/434 which contains more details.

Summary:
Adds Robot Framework test coverage for MarkLogic 12 dynamic-host management API behavior changes, including decoded-token fields, token deletion endpoints, and cross-cluster visibility/authorization rules.

Changes:

- Extend Dynamic Host Cluster Test with new steps for decoded token inspection and token deletion by JTI/host-id on port 7102.
- Add a new coupled-clusters test that couples two clusters and validates cross-cluster API behavior (GET empty, POST rejected, DELETE not-found).
- Introduce a new docker-compose scenario to spin up two independent 2-node clusters for the coupling test.

#### Checklist: 

-  ##### Owner:

- [x] JIRA_ID as part of branch/PR name
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  
- ##### Reviewer:

- [ ] Reviewed Tests
- [ ] Added to Release Wiki/Jira

### CODE REVIEW

File: test/keywords.resource [hunk 1]
```diff
@@ -693,4 +859,117 @@ Verify That marklogic.conf contains
...
+    ${foreign_cluster_id}=    Set Variable    ${props.json()["cluster-id"]}
+    ${foreign_cluster_name}=    Set Variable    ${props.json()["cluster-name"]}
+
+    # Get bootstrap host info - find host with 'bootstrap' in its hostname
+    ${hosts_resp}=    GET On Session    ForeignMgmt_${foreign_port}    url=/manage/v2/hosts?format=json    headers=${header_json}
+    ${hosts}=    Set Variable    ${hosts_resp.json()["host-default-list"]["list-items"]["list-item"]}
+    # Default to first host, then override if we find one named 'bootstrap'
+    ${foreign_host_id}=    Set Variable    ${hosts[0]["idref"]}
+    ${foreign_host_name}=    Set Variable    ${hosts[0]["nameref"]}
+    FOR    ${host}    IN    @{hosts}
```
[the remark this answers is not in the document]
Comment (tposham) [reply test/keywords.resource:884]: Not applicable for this test. `compose-test-17.yaml` always brings up 2 nodes per cluster (cluster1_bootstrap + cluster1_node2, cluster2_bootstrap + cluster2_node2), so /manage/v2/hosts always returns a list. The single-item dict edge case handled elsewhere in the file applies to single-node compose tests , this keyword is only called after a 2-node compose is fully up. No change needed.

File: test/keywords.resource [hunk 2]
```diff
@@ -693,4 +859,117 @@ Verify That marklogic.conf contains
...
+    ...    DELETE /dynamic-hosts/{real-host-id} → 404 Not Found (dynamic hosts not visible cross-cluster)
+    ${auth}    ${headers}=    Generate digest authorization for    ${DEFAULT ADMIN USER}    ${DEFAULT ADMIN PASS}
+    ${header}=    Create Dictionary    Accept=application/json
+    ${header_post}=    Create Dictionary    Content-type=application/json    Accept=application/json
+    ${response}=    Create Digest Session    CrossClusterSess_${local_port}    url=http://localhost:${local_port}    headers=${headers}    auth=${auth}
+
+    # Derive foreign management port: cluster2 = 7302, cluster1 = 7102
+    ${is_cluster2}=    Run Keyword And Return Status    Should Contain    ${foreign_cluster_name}    cluster2
+    ${foreign_mgmt_port}=    Set Variable If    ${is_cluster2}    7302    7102
+    ${foreign_session}=    Create Digest Session    ForeignSess_${foreign_mgmt_port}    url=http://localhost:${foreign_mgmt_port}    headers=${headers}    auth=${auth}
```
[the remark this answers is not in the document]
Comment (tposham) [reply test/keywords.resource:928]: Not a real risk for this test. MarkLogic derives the cluster name from the bootstrap hostname, so cluster1_bootstrap → cluster1_bootstrap-cluster and cluster2_bootstrap → cluster2_bootstrap-cluster. These names are fully controlled by `compose-test-17.yaml` which we own, and will always contain the expected substring. The keyword is purpose-built for this test's compose setup, not a generic utility. No change needed.

**Review by rwinieski — APPROVED** [review]
